src/lib/expandApplyAtRules.js

```diff
@@ -35,6 +37,94 @@ function prefix(context, selector) {
   return typeof prefix === 'function' ? prefix(selector) : prefix + selector
 }
 
+function* pathToRoot(node) {
+  yield node
+  while (node.parent) {
+    yield node.parent
+    node = node.parent
+  }
+}
+
+function shallowClone(node, overrides = {}) {
+  let children = node.nodes
+  node.nodes = []
+
+  const tmp = node.clone(overrides)
+
+  node.nodes = children
+
+  return tmp
+}
+
+/**
+ * @param {import('postcss').Node} node
+ */
+function structuralCloneOfNode(node) {
```

Comment:
I get what this is doing (cloning a node) but I think that it might be a bit misleading because if I read this correctly it also walks up the tree and eventually returns a the node (if it is already the root) or a parent of the node.

I don't immediately know a better name though 🤔

Reply:
Yeah, I couldn't think of a better name either. Would definitely welcome something better. It's purpose is to clone the singular, nested version of a node that can be used to represent it.

So, if we have the source:
```css
@media (min-width: 768px) {
  @supports (display: grid) {
    .foo {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
    }
  }

  @supports (backdrop-filter: blur(1px)) {
    .bar {
      backdrop-filter: blur(1px);
    }
  }

  .baz {
    color: orange;
  }
}
```

And we're currently looking at `.bar` it'll return a cloned version of what's required for just that single node:
```css
@media (min-width: 768px) {
  @supports (backdrop-filter: blur(1px)) {
    .bar {
      backdrop-filter: blur(1px);
    }
  }
}
```

Reply:
idk maybe `nestedClone`?

Reply:
Yeah just `nestedClone` is fine, not perfect but will do the job 👍

Reply:
Okay, I went with `nestedClone`. I still don't like it but it feels less ridiculous.